helix-term/src/ui/editor.rs

```diff
@@ -363,17 +386,33 @@ impl EditorView {
             if range.head > range.anchor {
                 // Standard case.
                 let cursor_start = prev_grapheme_boundary(text, range.head);
-                spans.push((selection_scope, range.anchor..cursor_start));
+                spans.push(syntax::Span {
+                    scope: Highlight(selection_scope),
+                    start: range.anchor,
+                    end: cursor_start,
+                });
```

Comment:
Some of the spans in this function will be zero-width - for example here when `range.anchor == cursor_start` (in a single-width selection, cursor == anchor). I was thinking we might want to wrap `spans.push` in a closure that doesn't push if `range.start == range.end`. What do you think?

Reply:
The `Overlay` iterator already skipts empty spans anyways so the extra check here would be kind of double. 
It might be a small performance improvement by avoiding extra allocations but I doubt that would be worth it.

I think a better improvement in the future would be to implement this as a proper lazy iterator to avoid the unneeded `Vec` allocations but that's a bit more effort and there are usually not that many cursors around so its probably not worth doing in this PR (maybe in a future PR instead)